pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -557,6 +565,39 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         applyNCMultiQubitOp(arr, num_qubits, matrix, {}, {}, wires, inverse);
     }
 
+    template <class PrecisionT>
+    static void
+    applyPauliRot(std::complex<PrecisionT> *arr, std::size_t num_qubits,
+                  const std::vector<std::size_t> &controlled_wires,
+                  const std::vector<bool> &controlled_values,
+                  const std::vector<std::size_t> &wires, const bool inverse,
+                  PrecisionT angle, const std::string &word) {
+        constexpr std::size_t one{1};
+        constexpr auto IMAG = Pennylane::Util::IMAG<PrecisionT>();
+        const std::size_t n_wires = wires.size();
+        const std::size_t dim = one << n_wires;
+        const PrecisionT c = std::cos(angle / 2);
+        const std::complex<PrecisionT> s =
+            ((inverse) ? IMAG : -IMAG) * std::sin(angle / 2);
```

Comment:
```suggestion
            ((inverse) ? IMAG : -IMAG) * std::sin(angle / 2.0);
```
To be sure to prevent recast